Week_2/functions/palindromFilter/palindromFilter.cpp

```diff
@@ -0,0 +1,39 @@
+#include <string>
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+bool IsPalindrom(string str)
+{
+    string reversed_str = str;
+    reverse(str.begin(), str.end());
+    return reversed_str == str;
+}
+
+vector<string> PalindromFilter(const vector<string> &input, int length)
+{
+    vector<string> result;
+    for (const string &word : input)
+    {
+        if (IsPalindrom(word) && word.size() > length)
```

Comment:
Palindrom is heavy function in compare to size(). You should firstly check size and only after call palindrom. Because if `word.size() > length` is false `IsPalindrom(word)` won't be called at all.

Reply:
in JS we never think about memory, but sounds reasonable